### PR TITLE
[fix][sf] fix bug with PyPredictor to remove worker, add specific fla…

### PR DESCRIPTION
**Pull request by siddvenk**
…g for sm health check behavior

## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: serving/src/main/java/ai/djl/serving/models/ModelManager.java [hunk 1]
```diff
@@ -416,6 +417,18 @@ public CompletableFuture<Map<String, Object>> workerStatus() {
                                         if (wlm.getWorkerPool(wpc).isFullyScaled()) {
                                             data.put(modelName, new StatusResponse("Healthy"));
                                         } else {
+                                            boolean sageMakerHealthCheckOverride =
```
Comment (frankfliu) [inline serving/src/main/java/ai/djl/serving/models/ModelManager.java:420]:
```suggestion
if (Boolean.getBoolean("SAGEMAKER_HEALTH_CHECK_OVERRIDE")) {
```

File: serving/src/main/java/ai/djl/serving/models/ModelManager.java [hunk 2]
```diff
@@ -416,6 +417,18 @@ public CompletableFuture<Map<String, Object>> workerStatus() {
...
+                                                    Boolean.parseBoolean(
+                                                            Utils.getenv(
+                                                                    "SAGEMAKER_HEALTH_CHECK_OVERRIDE",
+                                                                    "false"));
+                                            if (sageMakerHealthCheckOverride) {
+                                                logger.warn(
```
Comment (frankfliu) [inline serving/src/main/java/ai/djl/serving/models/ModelManager.java:426]: This should be info level

File: serving/src/main/java/ai/djl/serving/models/ModelManager.java (hunk 1 above)
Comment (siddvenk) [reply serving/src/main/java/ai/djl/serving/models/ModelManager.java]: updated

File: serving/src/main/java/ai/djl/serving/models/ModelManager.java (hunk 2 above)
Comment (siddvenk) [reply serving/src/main/java/ai/djl/serving/models/ModelManager.java]: updated, and also changed the other log we had for SERVING_HEALTH_CHECK_OVERRIDE from error -> info

**Review by frankfliu — APPROVED** [review]

**Review by sindhuvahinis — APPROVED** [review]

**Pull request merged**